src/native-logger-listener.ts

```diff
@@ -1,52 +1,67 @@
-import { NativeEventEmitter, NativeModules, Platform } from 'react-native';
+import { NativeEventEmitter } from 'react-native';
+import NativeCustomerIOLogging from './specs/modules/NativeCustomerIOLogging';
 import { CioLogLevel } from './types';
+import { callNativeModule, ensureNativeModule } from './utils/native-bridge';
 
-const LINKING_ERROR =
-  `The package 'customerio-reactnative' doesn't seem to be linked. Make sure: ` +
-  Platform.select({ ios: "- You have run 'pod install'\n", default: '' }) +
-  '- You rebuilt the app after installing the package\n' +
-  '- You are not using Expo Go\n';
+// Reference to the native CustomerIO logging module for SDK operations
+const nativeModule = ensureNativeModule(NativeCustomerIOLogging);
 
-const CioLoggingEmitter = NativeModules.CioLoggingEmitter
-  ? NativeModules.CioLoggingEmitter
-  : new Proxy(
-      {},
-      {
-        get() {
-          throw new Error(LINKING_ERROR);
-        },
-      }
-    );
+// Wrapper function that ensures SDK is initialized before calling native methods
+const withNativeModule = <R>(
+  fn: (native: typeof NativeCustomerIOLogging) => R
+): R => {
+  return callNativeModule(nativeModule, fn);
+};
 
 export class NativeLoggerListener {
-  static initialize() {
-    const bridge = new NativeEventEmitter(CioLoggingEmitter);
+  private static isInitialized = false;
+
+  static async initialize() {
+    // Prevent multiple registrations of the same log listener
+    if (this.isInitialized) {
+      return;
+    }
     const loggerPrefix = '[CIO] ';
-    bridge.addListener(
-      'CioLogEvent',
-      (event: { logLevel: CioLogLevel; message: string }) => {
-        // Using console.log will log to the JavaScript side but prevent
-        // React Native’s default behavior of redirecting logs to the native side.
-        // By doing it asynchronously, we ensure the logs are captured on both
-        // the JavaScript and native ends.
-        async function log() {
-          switch (event.logLevel) {
-            case CioLogLevel.Debug:
-              console.debug(loggerPrefix + event.message);
-              break;
-            case CioLogLevel.Info:
-              console.info(loggerPrefix + event.message);
-              break;
-            case CioLogLevel.Error:
-              console.error(loggerPrefix + event.message);
-              break;
-            default:
-              console.log(loggerPrefix + event);
-              break;
-          }
+
+    const logHandler = (data: any) => {
+      const event = data as { logLevel: CioLogLevel; message: string };
+      // Using console.log will log to the JavaScript side but prevent
+      // React Native's default behavior of redirecting logs to the native side.
+      // By doing it asynchronously, we ensure the logs are captured on both
+      // the JavaScript and native ends.
+      async function log() {
```

Comment:
we are declaring it on every event, cant we define it once instead of multiple closures?